jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/OdeTimJsonTopology.java

```diff
@@ -11,51 +11,64 @@
- * K-Table that houses TMC-generated TIMs which can be queried by UUID.
- **/
+ * OdeTimJsonTopology sets up and manages a Kafka Streams topology for
+ * processing TIM JSON data from a specified Kafka topic.
+ * This class filters incoming TIM messages to retain only those generated by
+ * the TMC, storing them in a KTable keyed by each message's UUID.
+ * This class provides a query method to retrieve and automatically remove (via
+ * tombstone) a TIM message by its UUID, ensuring the KTable remains up-to-date
+ * and free of stale entries.
+ * The topology is configured using provided Kafka properties and supports both
+ * standard and Confluent Kafka deployments.
+ */
 @Slf4j
 public class OdeTimJsonTopology {
 
   private final KafkaStreams streams;
+  private final KafkaTemplate<String, String> tombstonePublisher;
+
+  private String topic;
 
   /**
-   * Constructs an instance of OdeTimJsonTopology to set up and manage a Kafka Streams topology for
-   * processing TIM JSON data.
+   * Constructs an instance of OdeTimJsonTopology to set up and manage a Kafka
+   * Streams topology for processing TIM JSON data.
    *
-   * @param odeKafkaProps the properties containing Kafka configuration, including brokers and
-   *                      optional Confluent-specific configuration for authentication.
-   * @param topic         the Kafka topic from which TIM JSON data is consumed to build the
-   *                      topology.
+   * @param odeKafkaProps the properties containing Kafka configuration, including
+   *                      brokers and optional Confluent-specific configuration
+   *                      for authentication.
+   * @param topic         the Kafka topic from which TIM JSON data is consumed to
+   *                      build the topology.
+   * @param template      The KafkaTemplate used to publish tombstone messages to
+   *                      the topic.
    */
-  public OdeTimJsonTopology(OdeKafkaProperties odeKafkaProps, String topic) {
+  public OdeTimJsonTopology(OdeKafkaProperties odeKafkaProps, String topic,
+      KafkaTemplate<String, String> template) {
+    // The container name in Docker or pod name in Kubernetes
+    String hostname = System.getenv("HOSTNAME") != null ? "-" + System.getenv("HOSTNAME") : "";
 
     Properties streamsProperties = new Properties();
-    streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, "KeyedOdeTimJson");
     streamsProperties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, odeKafkaProps.getBrokers());
     streamsProperties.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
     streamsProperties.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class);
-    streamsProperties.put(StreamsConfig.WINDOW_SIZE_MS_CONFIG, 3600 * 1000L); // 1 hour retention
+    streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, "ode-tim-json-topology" + hostname);
```

Comment:
suggestiion: Use a global Ktable instead.

Reply:
I have now reworked the solution to utilize the global ktable.